abci/types/application.go

```diff
@@ -17,11 +17,13 @@ type Application interface {
 	CheckTx(RequestCheckTx) ResponseCheckTx // Validate a tx for the mempool
 
 	// Consensus Connection
-	InitChain(RequestInitChain) ResponseInitChain    // Initialize blockchain w validators/other info from TendermintCore
+	InitChain(RequestInitChain) ResponseInitChain // Initialize blockchain w validators/other info from TendermintCore
+	PrepareProposal(RequestPrepareProposal) ResponsePrepareProposal
 	BeginBlock(RequestBeginBlock) ResponseBeginBlock // Signals the beginning of a block
 	DeliverTx(RequestDeliverTx) ResponseDeliverTx    // Deliver a tx for full processing
 	EndBlock(RequestEndBlock) ResponseEndBlock       // Signals the end of a block, returns changes to the validator set
 	Commit() ResponseCommit                          // Commit the state and return the application Merkle root hash
+	ProcessProposal(RequestProcessProposal) ResponseProcessProposal
```

Comment:
Can we move this up to PrepareProposal and also a doc comment like everywhere else?

Reply:
nice find, yeah totally https://github.com/celestiaorg/celestia-core/pull/737/commits/79b95615e795864cf0fdfc67a3f3afba2088bab7

I made them slightly celestia specific by mentioning block data, but there are no comments yet on upstream master, so we can switch to those whenever they are added